src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -283,37 +283,54 @@ static bool FastRegisterNativeMembers (JniType nativeClass, Type type, string me
 		}
 
 		class MagicRegistrationMap {
-			static Dictionary<string, int> typesMap;
-
-			static void Prefill ()
-			{
-				// fill code added by the linker
-			}
+			// should stay in sync with MonoDroidMarkStep.GetStringHashCode
```

Comment:
Instead of a comment, why don't we stick this into a dedicated `.cs` file which is included into both projects, possibly via (new?) shared project?  Then we don't need to document that the methods be in sync, as the same source will be used.